wgpu-core/src/device/global.rs

```diff
@@ -2435,6 +2510,16 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
         }
     }
 
+    pub fn device_lose<A: HalApi>(&self, device_id: DeviceId) {
+        let hub = A::hub(self);
+        let mut token = Token::root();
+
```

Comment:
should leave todo note on all the other things we're missing (pretty much from the spec):
bunch of other todos that we should note down, pretty much verbatim from the spec:

* Resolve device.lost promise (once implemented)
* Complete any outstanding mapAsync
* Complete any outstanding onSubmittedWorkDone

Reply:
I put this comment in the implementation of `lose` in resource.rs.